crates/storage/src/connection.rs

```diff
@@ -108,4 +116,12 @@ impl Transaction<'_> {
     pub fn trie_pruning_enabled(&self) -> bool {
         matches!(self.trie_prune_mode, TriePruneMode::Prune { .. })
     }
+
+    /// Resets the [`Storage`](crate::Storage) state. Required after each reorg.
+    pub fn reset(&self) -> anyhow::Result<()> {
+        self.rebuild_running_event_filter()?;
```

Comment:
How do we handle the case where after a reorg the number of blocks for which the aggregate bloom filter is missing is larger than the size of an aggregate filter?

Reply:
Hmm, I think you're right.